File: solutions/opus-1/opus/python/primes.py
import sys

def main():
    data = sys.stdin.read().split()
    n = int(data[0]) if data else 0
    if n < 2:
        sys.stdout.write("\n")
        return
    sieve = bytearray([1]) * (n + 1)
    sieve[0] = sieve[1] = 0
    i = 2
    while i * i <= n:
        if sieve[i]:
            sieve[i * i::i] = bytearray(len(range(i * i, n + 1, i)))
        i += 1
    sys.stdout.write(" ".join(str(k) for k in range(n + 1) if sieve[k]) + "\n")

main()

File: solutions/opus-1/opus/rust/primes.rs
use std::io::{self, Read, Write};

fn main() {
    let mut s = String::new();
    io::stdin().read_to_string(&mut s).unwrap();
    let n: usize = s.trim().parse().unwrap_or(0);
    let mut sieve = vec![true; n + 1];
    let mut out = Vec::new();
    for i in 2..=n {
        if sieve[i] {
            out.push(i.to_string());
            let mut j = i * i;
            while j <= n {
                sieve[j] = false;
                j += i;
            }
        }
    }
    let mut stdout = io::stdout();
    writeln!(stdout, "{}", out.join(" ")).unwrap();
}
